tests/Domain/Integrations/Repositories/EloquentIntegrationRepositoryTest.php

```diff
@@ -248,6 +248,7 @@ public function test_it_can_activate_with_a_coupon(): void
 
         $this->integrationRepository->save($searchIntegration);
 
+        $this->withoutEvents();
```

Comment:
Can this not be solved here https://github.com/cultuurnet/publiq-platform/blob/bc34e139efbafb81de7e192b5868518fb1511063/tests/TestCase.php#L19-L21 ?

Reply:
FYI I think you can just do `Model::unsetEventDispatcher()` once because it's a static method defined on `Model`

Reply:
The tests complain about it: `No query results for model [App\Insightly\Models\InsightlyMappingModel` Though maybe I could look to fix it with a `Mock`

Reply:
I think you should not use `Model::unsetEventDispatcher();` but `IntegrationModel::unsetEventDispatcher();`

Reply:
I can even run the tests locally when I remove the line with `$this->withoutEvents();` 🤔

Reply:
`Model::unsetEventDispatcher()` disables it for all models. See https://github.com/laravel/framework/issues/29727
So that should be sufficient to replace those 3 separate calls to `unsetEventDispatcher()`.
But that comment was just about those 3 specific existing lines, not about your question about `withoutEvents()`.

Reply:
The issue is something totally different. In the base class `TestCase` we did put the `unsetEventDispatcher` because we wanted to prevent the dispatch from being called after "normal" Eloquent model interaction.

But now we fire do `dispatch` inside a custom method so that will not work https://github.com/cultuurnet/publiq-platform/blob/4d3647afacf18175babfd03d259e42f629eeb832/app/Domain/Integrations/Models/IntegrationModel.php#L56

Reply:
The solution is to use the trait `WithoutEvents`. This trait will be executed inside the `setUp` method.

The disabling of events has to happen exactly at that code location because we have 2 distinct ways of using the `dispatch`. The first and common approach is inside the static `booted` method and the new approach is inside the `activeWithCoupon` method.

If we disable the events later, for example in our custom `setUp` method then the `dispatch` inside the `booted` method are still done.

Reply:
Solved in https://github.com/cultuurnet/publiq-platform/pull/111/commits/61e5e015ee961605d7c397ed7b656508e41db81a

Reply:
Magic 🪄